protocol/protocol.go

```diff
@@ -28,8 +28,7 @@ func (m *Manager) routerInitializer(netConnection *netadapter.NetConnection) (*r
 		if err != nil {
 			if protocolErr := &(protocolerrors.ProtocolError{}); errors.As(err, &protocolErr) {
 				if protocolErr.ShouldBan {
-					// TODO(libp2p) Ban peer
-					panic("unimplemented")
+					m.context.ConnectionManager().Ban(netConnection)
```

Comment:
This is the only place where you call Ban.
And netAdapter is available here.
And ConnectionManager().Ban does NetAdapter.Ban()

So remove ConnectionManager.Ban and call NetAdapter.Ban() here?

Reply:
We spoke about it and decided to leave it as is for now.